packages/devtools_app/lib/src/shared/preferences.dart

```diff
@@ -268,3 +279,57 @@ class InspectorPreferencesController extends DisposableController
     _hoverEvalMode.value = enableHoverEvalMode;
   }
 }
+
+class MemoryPreferencesController {
+  ValueListenable<bool> get androidCollectionEnabled =>
+      _androidCollectionEnabled;
+  final _androidCollectionEnabled = ValueNotifier<bool>(false);
+  static const _androidCollectionEnabledStorageId =
+      'memory.androidCollectionEnabled';
+
+  ValueListenable<bool> get autoSnapshotEnabled => _autoSnapshotEnabled;
+  final _autoSnapshotEnabled = ValueNotifier<bool>(false);
+  static const _autoSnapshotEnabledStorageId = 'memory.autoSnapshotEnabled';
+
+  Future<void> init() async {
+    final androidCollectionEnabledValue =
+        await storage.getValue(_androidCollectionEnabledStorageId);
+    setAndroidCollectionEnabled(androidCollectionEnabledValue == 'true');
+    _androidCollectionEnabled.addListener(() {
```

Comment:
we should probably auto dispose the listeners here too

Reply:
+1 use auto dispose mixin functionality